### PR TITLE
Create the profile page

### PR DESCRIPTION
**Pull request by nlitwin**
## Description

**Issue:** #78 

Creating the profile page, which consists of two large sections, the user's details and the user's recent posts.

## Subtasks

- [x] I have added this PR to the Journaly Kanban project ✅
- [x] Profile Card
  - [x] Responsive 
  - [x] background image
- [x] Recent Posts
- [x] Flesh out Post model definition so we can use the Post type on the frontend
- [x] Support querying posts by `userId`

### Desktop

dashboard_profile_page

### CODE REVIEW

**Comment by robin-macpherson** [comment]
@nlitwin looking greaaaat so far mate!!

File: README.md [hunk 1]
```diff
@@ -39,42 +39,42 @@ We use Python for data science, leveraging a number of powerful libraries for da
...
-$ brew install postgres
-$ brew services start postgresql
-```
+   ```bash
+   $ brew install postgres
+   $ brew services start postgresql
+   ```
 
-2. Start up your Postgres shell and create your user
+1. Start up your Postgres shell and create your user
```
Comment (nlitwin) [inline README.md:55]: Little markdown trick so you can reorder and add bullet points without changing the numbers around. It will still increment when it's rendered

File: README.md [hunk 2]
```diff
@@ -39,42 +39,42 @@ We use Python for data science, leveraging a number of powerful libraries for da
...
-$ create database journaly_db;
-$ alter user <your_username> with superuser;
-```
+   ```bash
+   $ create database journaly_db;
+   $ alter user <your_username> with superuser;
+   ```
 
-4. Apply database migrations to your new database instance:
+1. If you haven't created a `backend/prisma/.env` file yet, copy and paste `backend/prisma/.env.example` into the new `.env` file. Then, replace the placeholders with your new postgres username & password.
```
Comment (nlitwin) [inline README.md:69]: Prisma was yelling at me in the console that my `.env` file wasn't in the `prisma` directory. did you run into this too?

File: README.md [hunk 3]
```diff
@@ -39,42 +39,42 @@ We use Python for data science, leveraging a number of powerful libraries for da
...
 
-```bash
-$ npx prisma migrate up --experimental
-```
+1. Finally, from the `backend` directory, apply database migrations to your new database instance:
 
-5. Finally, go to your `.env` file in the backend directory and replace the placeholders with your new postgres username & password.
+   ```bash
+   $ npx @prisma/cli migrate up --experimental
```
Comment (nlitwin) [inline README.md:74]: The only way I got this command to work was with `@prisma/cli` as opposed to `prisma`

File: backend/.env.example [hunk 4]
```diff
@@ -1 +1,2 @@
-DATABASE_URL='postgresql://<your_username>:<your_password>@localhost:5432/journaly_db'
+FRONTEND_URL='http://localhost:3000'
```
Comment (nlitwin) [inline backend/.env.example:1]: I didn't have these 2 values in any .env file, but they were required to run the server, so I added them to the example file

File: backend/src/graphql.ts [hunk 5]
```diff
@@ -55,10 +56,12 @@ schema.objectType({
     t.model.title()
     t.model.body()
     t.model.excerpt()
+    t.model.readTime()
     t.model.author()
     t.model.status()
     t.model.threads()
     t.model.language({ type: 'Language' })
+    t.string('createdAt')
```
Comment (nlitwin) [inline backend/src/graphql.ts:64]: I couldn't figure out how to add `createdAt` as a regular `t.model.X` or "DateTime", nor could I find docs :( Right now this is working but I think preferably we'd send `createdAt` as a number / DateTime

File: backend/src/utils.ts [hunk 6]
```diff
@@ -11,7 +11,7 @@ const typeToElStrMap = {
   'heading-two': 'h2',
   'bulleted-list': 'ul',
   'numbered-list': 'ol',
-  'paragraph': 'p',
+  paragraph: 'p',
```
Comment (nlitwin) [inline backend/src/utils.ts:14]: Prettier does this automatically since regular keys don't need quotes, but keys with dashes need quotes

File: backend/src/utils/post.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+// TODO: enhance accuracy by taking into account HTML tags (only look at text and disregard HTML)
+export const readTime = (text: string): number => {
```
Comment (nlitwin) [inline backend/src/utils/post.ts:2]: Added this simple function for now since we'll need to calculate `readTime` on the backend. To be more accurate, we'll want to count characters instead of words potentially as well as stripping out HTML tags

File: frontend/codegen.yml [hunk 8]
```diff
@@ -3,6 +3,9 @@ schema: 'http://localhost:4000/graphql'
 documents: 'graphql/**/*.graphql'
 generates:
   generated/graphql.tsx:
+    hooks:
```
Comment (nlitwin) [inline frontend/codegen.yml:6]: Automatically runs prettier on our codegen file so `generated/graphql.tsx` doesn't change 🕺

File: frontend/components/Dashboard/PostCard/MyFeedPostCard.tsx [hunk 9]
```diff
@@ -1,32 +1,34 @@
...
+// TODO: use Post type from generated graphql
+import { Post } from './postCardUtils'
 
-interface Props {
-  id: string
-  title: string
-  body: string
-  image?: string
-  author: string
+type Props = {
```
Comment (nlitwin) [inline frontend/components/Dashboard/PostCard/MyFeedPostCard.tsx:6]: Minor updates in this file, but this component, which is the post that appears in the my feed page, needs more work

File: frontend/components/Dashboard/PostCard/postCardUtils.ts [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+  author: User
+  status: PostStatus
+  likes?: number
+  threads?: Thread[]
+  createdAt: string
+}
+
+export const postBorderRadius = '5px'
+
+export const processPost = (post: Post) => {
```
Comment (nlitwin) [inline frontend/components/Dashboard/PostCard/postCardUtils.ts:22]: Since we're creating separate components for posts that appear in various pages, this function helps consume the API response of a post and turn it into something that every Post component can use

File: frontend/components/Layouts/DashboardLayout.tsx [hunk 11]
```diff
@@ -1,10 +1,89 @@
...
+import Header from '../Dashboard/Header'
+import useWindowSize from '../../hooks/useWindowSize'
 
 interface Props {
   children: React.ReactNode
+  withPadding?: boolean
 }
 
-const DashboardLayout: React.FC<Props> = ({ children }) => <Dashboard>{children}</Dashboard>
+const DashboardLayout: React.FC<Props> = ({ children, withPadding = true }) => {
```
Comment (nlitwin) [inline frontend/components/Layouts/DashboardLayout.tsx:14]: I noticed that the `Dashboard` component was really the `DashboardLayout` component and `DashboardLayout` was a just a wrapper (see line 8 on the left), so I moved it from `Dashboard` to `DashboardLayout`

File: frontend/elements/ExternalLink/ExternalLink.tsx [hunk 12]
```diff
@@ -0,0 +1,48 @@
+import React from 'react'
```
Comment (nlitwin) [inline frontend/elements/ExternalLink/ExternalLink.tsx:1]: Helper component for anchor tags that go to a site outside of journaly.com

File: frontend/package.json [hunk 13]
```diff
@@ -38,7 +36,8 @@
     "react-hook-form": "^5.7.2",
     "slate": "^0.57.1",
     "slate-history": "^0.57.1",
-    "slate-react": "^0.57.1"
+    "slate-react": "^0.57.1",
+    "xss": "^1.0.7"
```
Comment (nlitwin) [inline frontend/package.json:40]: I replaced `dompurify` with `xss` since `dompurify` isn't isomorphic and requires a window (I kept getting errors in the console when SSR code tried to run `DOMPurify.sanitize`. We could add wrapper code to handle this, but I think it's simpler to just use a library that works out of the box on client or server

File: frontend/utils/index.ts [hunk 14]
```diff
@@ -10,3 +11,7 @@ export function fromEntries<V>(iterable: Iterable<[string, V]>) {
     return obj
   }, {} as { [k: string]: V })
 }
+
+export const sanitize = (html: string): string => {
```
Comment (nlitwin) [inline frontend/utils/index.ts:15]: Created this util function so that we can swap out the implementation with another library in just one place if we ever need to

File: README.md [hunk 15]
```diff
@@ -39,42 +39,42 @@ We use Python for data science, leveraging a number of powerful libraries for da
...
+   ```
 
-You've got a local Journaly PostgreSQL DB, woohoo! 🎉
+   You've got a local Journaly PostgreSQL DB, woohoo! 🎉
```
Comment (robin-macpherson) [inline README.md:77]: Still not understanding the indentation here? Is it GitHub diffs being weird or?

File: README.md [hunk 16]
```diff
@@ -83,11 +83,8 @@ You've got a local Journaly PostgreSQL DB, woohoo! 🎉
 ### Running Journaly
 
 1. To run the entire app in local development mode, simply run `npm run dev` from the root of the project! This will use `concurrently` to start up both the frontend and backend dev servers.
-2. If you would like to run frontend or backend separately, you can run `npm run dev` from each respective directory.
-
-### Seeding your database
-
-3. Let's seed that baby DB!  
+1. If you would like to run frontend or backend separately, you can run `npm run dev` from each respective directory.
+1. Let's seed that baby DB!
```
Comment (robin-macpherson) [inline README.md:87]: Since you're updating these let's update the docs so that they recommend running `npm run reseed-db` from the project root now

File: backend/src/utils/post.ts (hunk 7 above)
Comment (robin-macpherson) [reply backend/src/utils/post.ts:2]: Cool noice! 😄

File: frontend/components/Dashboard/PostCard/MyFeedPostCard.tsx [hunk 17]
```diff
@@ -69,4 +71,4 @@ const PostCard: React.FC<Props> = ({ id, title, body, image, author }) => {
   )
 }
 
-export default PostCard
+export default MyFeedPostCard
```
Comment (robin-macpherson) [inline frontend/components/Dashboard/PostCard/MyFeedPostCard.tsx:74]: I wonder if we should give these slightly more generic names, just in case their purpose changes in the future, such as `PostCardStack` for the one where the info is stacked vertically in the feed currently and `PostCardInline` or something along those lines. We could use flexbox naming conventions of `PostCardCol` and `PostCardRow`. That feels like it would be better to me, since these could easily one day be repurposed and they are quite flexible, so giving them page-specific names feels a wee bit wonky, but let me know what you think 😊

File: frontend/components/Dashboard/PostCard/postCardUtils.ts [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+  const displayImage = images[0] || '/images/samples/sample-post-img.jpg'
+  const authorDisplayName = author.name || author.handle
+
+  return {
+    id,
+    title,
+    readTime,
+    likes,
+    numThreads: threads.length,
+    excerpt: sanitize(post.excerpt),
```
[the remark this answers is not in the document]
Comment (robin-macpherson) [reply frontend/components/Dashboard/PostCard/postCardUtils.ts:33]: Yep I agree that here we will just be grabbing text content. We can write something really nice later than calculates what we can grab based on the number of chars in each element and how many max chars we can fit and so on 😄

File: frontend/components/Dashboard/PostCard/postCardUtils.ts (hunk 10 above)
Comment (robin-macpherson) [reply frontend/components/Dashboard/PostCard/postCardUtils.ts:22]: I'm not sure if I fully understand the need for this. Can you explain a bit more the problem we're solving?

File: frontend/components/Dashboard/PostCard/postCardUtils.ts [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+    displayImage,
+    profileImage: author.profileImage,
+    authorName: authorDisplayName,
+    createdAt,
+  }
+}
+
+export const formatPostDate = (date: string): string => {
+  return format(Number(date), 'MMM d')
+}
```
Comment (robin-macpherson) [inline frontend/components/Dashboard/PostCard/postCardUtils.ts:43]: Thanks for making that!

File: frontend/components/Dashboard/Profile/PostList.tsx [hunk 20]
```diff
@@ -0,0 +1,57 @@
+import React from 'react'
+import { useTranslation } from '../../../config/i18n'
+import { layoutLeftRightPadding } from '../../Dashboard/dashboardConstants'
+import ProfilePostCard from '../PostCard/ProfilePostCard'
+import theme from '../../../theme'
+
+type Props = {
+  // TODO: use Post type from generated graphql
```
Comment (robin-macpherson) [inline frontend/components/Dashboard/Profile/PostList.tsx:8]: This may be fixed now since we've improved all the backend stuff a ton. It's worth giving it a try since it came up in this PR a few times

File: frontend/components/Dashboard/PostCard/postCardUtils.ts [hunk 21]
```diff
@@ -0,0 +1,43 @@
+import { format } from 'date-fns'
+import { sanitize } from '../../../utils'
+import { User, PostStatus, Thread } from '../../../generated/graphql'
+
+// TODO: this will be unnecessary once we use types from generated graphql
```
[the remark this answers is not in the document]
Comment (robin-macpherson) [reply frontend/components/Dashboard/PostCard/postCardUtils.ts:5]: Yep we're already using the generated GraphQL types. I just had trouble in the past with the `Post` type in particular, as it had been generated incorrectly somehow, but this may be fixed now since we've improved all the backend stuff a ton. It's worth giving it a try since it came up in this PR a few times

File: frontend/components/Dashboard/Profile/ProfileCard.tsx [hunk 22]
```diff
@@ -0,0 +1,244 @@
...
+  }
+
+  const name = user.name || user.handle
+  const showSeparator = user.facebook || user.instagram || user.youtube || user.website
+
+  return (
+    <div className="profile-card">
+      <div className="profile-header">
+        <h1 className="profile-name">{name}</h1>
+
```
Comment (robin-macpherson) [inline frontend/components/Dashboard/Profile/ProfileCard.tsx:36]: Is there any reason for the empty lines here and there throughout the file amongst the markup?

File: frontend/utils/index.ts (hunk 14 above)
Comment (robin-macpherson) [reply frontend/utils/index.ts:15]: Nice! 🎉

File: README.md [hunk 23]
```diff
@@ -39,42 +39,42 @@ We use Python for data science, leveraging a number of powerful libraries for da
...
    _Note that this set your Postgres DB to run when your computer starts up and will stay running in the background_.
 
-```bash
-$ brew install postgres
-$ brew services start postgresql
-```
+   ```bash
+   $ brew install postgres
+   $ brew services start postgresql
+   ```
```
[the remark this answers is not in the document]
Comment (nlitwin) [reply README.md:53]: It's so that they line up with the content of each bullet point and the bullet points auto-increment:

image

File: README.md (hunk 15 above)
Comment (nlitwin) [reply README.md:77]: Same as comment above :D It's simply so the content of each bullet point lines up. Just in case you haven't seen this before, this little icon is handy if you ever need to see how the markdown changes render:

image

File: README.md (hunk 3 above)
Comment (nlitwin) [reply README.md:74]: Cool, thanks! I'll update the docs with the `npm run migrate:up` command 👍

File: frontend/components/Dashboard/PostCard/postCardUtils.ts (hunk 18 above)
[the remark this answers is not in the document]
Comment (nlitwin) [reply frontend/components/Dashboard/PostCard/postCardUtils.ts:33]: ok awesome! that simplifies things a bit 😄 I'll update it

File: frontend/components/Dashboard/Profile/ProfileCard.tsx (hunk 22 above)
Comment (nlitwin) [reply frontend/components/Dashboard/Profile/ProfileCard.tsx:36]: I almost always add empty lines between sibling elements unless it's very similar and repeated (like the `<p>` tags below and `social-link`s). It was something we did when I was at Topcoder and I've just continued doing it

File: frontend/components/Dashboard/PostCard/postCardUtils.ts (hunk 21 above)
[the remark this answers is not in the document]
Comment (nlitwin) [reply frontend/components/Dashboard/PostCard/postCardUtils.ts:5]: I will give this a try again shortly! But I believe when I used the generated graphql type, the Post didn't contain several attributes that I needed (`excerpt`, `readTime`, `likes`, etc.)

File: backend/src/graphql.ts [hunk 24]
```diff
@@ -117,114 +132,123 @@ schema.objectType({
...
-    t.string('type', { nullable: true }),
-      t.string('text', { nullable: true }),
-      t.boolean('italic', { nullable: true }),
-      t.boolean('bold', { nullable: true }),
-      t.field('children', {
-        type: EditorNode,
-        list: true,
-        nullable: true,
-      })
+    t.string('type', { nullable: true })
```
Comment (nlitwin) [inline backend/src/graphql.ts:135]: There were some extra commas on the end of these lines, causing prettier to indent it weird, so that's why there are lots of changes to this file

File: backend/src/graphql.ts [hunk 25]
```diff
@@ -117,114 +132,123 @@ schema.objectType({
...
+        authorId: intArg({ required: true }),
+      },
+      resolve: async (_root, args, ctx) => {
+        return ctx.db.post.findMany({
+          where: {
+            author: { id: args.authorId },
+          },
+        })
+      },
+    })
```
Comment (nlitwin) [inline backend/src/graphql.ts:161]: Lines 151-here are new though and aren't due to the prettier formatting changes

File: frontend/components/Dashboard/PostCard/MyFeedPostCard.tsx (hunk 17 above)
Comment (nlitwin) [reply frontend/components/Dashboard/PostCard/MyFeedPostCard.tsx:78]: Chatted in person about the approach - new post component at `frontend/components/Dashboard/Posts/Post.tsx` and is currently used on the profile page. I'll update the My Feed page once the `<Post>` component supports avatars in a future PR 😄

File: frontend/components/Dashboard/Posts/Post.tsx [hunk 26]
```diff
@@ -0,0 +1,172 @@
...
+    images,
+    likes,
+    threads,
+    author: { handle, name, profileImage },
+    createdAt,
+  } = post
+
+  const displayImage = images.length ? images[0].smallSize : '/images/samples/sample-post-img.jpg'
+  console.log(handle, name, profileImage, avatar)
+  // const authorDisplayName = author.handle || author.name
```
Comment (nlitwin) [inline frontend/components/Dashboard/Posts/Post.tsx:32]: Avatar/author section to come in a future PR

File: backend/src/graphql.ts (hunk 25 above)
Comment (robin-macpherson) [reply backend/src/graphql.ts:161]: I wonder if the `authorId` should be optional here 🤔 My thought earlier was more to extend this query to be able to serve this use case. What do you think?

File: frontend/components/Dashboard/PostCard/MyFeedPostCard.tsx (hunk 17 above)
Comment (robin-macpherson) [reply frontend/components/Dashboard/PostCard/MyFeedPostCard.tsx:79]: Oh I feel like that should perhaps be called `PostCard` then like before (the new one I mean). `Post` feels too generic to me and we already have a `Post` component for an entire post that someone is viewing, which feels more correct to me 🤔 What was wrong with `PostCard`?

File: frontend/components/Dashboard/Posts/Post.tsx [hunk 27]
```diff
@@ -0,0 +1,172 @@
...
+    readTime,
+    images,
+    likes,
+    threads,
+    author: { handle, name, profileImage },
+    createdAt,
+  } = post
+
+  const displayImage = images.length ? images[0].smallSize : '/images/samples/sample-post-img.jpg'
+  console.log(handle, name, profileImage, avatar)
```
Comment (robin-macpherson) [inline frontend/components/Dashboard/Posts/Post.tsx:31]: `console.log`

File: frontend/components/Dashboard/Profile/ProfileCard.tsx (hunk 22 above)
Comment (robin-macpherson) [reply frontend/components/Dashboard/Profile/ProfileCard.tsx:36]: Hmm I see, I feel like this might be the kind of thing that would be better either as a convention or not. It's definitely not like a huge thing of course but if each person has a very particular style preference then we end up with lots of files looking quite different from each other depending on who wrote them and at SS you'll often get people who end up submitting PRs where they have an auto-formatter set up one way and it changes a bunch of stuff but then the next person comes along and re-formats it their way 😂 I feel like readability of a codebase is very important and consistency is important there

File: frontend/graphql/feed.graphql [hunk 28]
```diff
@@ -3,10 +3,24 @@ query feed {
...
+    }
     author {
       id
+      handle
       name
       email
+      profileImage
     }
   }
 }
```
Comment (robin-macpherson) [inline frontend/graphql/feed.graphql:26]: Ideally we should try to use the fragments, you did update one in `fragments.graphql` but it looks like not using it anywhere here

File: frontend/pages/post/[id].tsx [hunk 29]
```diff
@@ -22,10 +22,14 @@ const PostPage: NextPage = () => {
       <LoadingWrapper
         loading={loading}
         error={error}
-        render={() => <Post post={post} refetch={refetch} />}
+        render={() => <NewPost post={post} refetch={refetch} />}
```
Comment (robin-macpherson) [inline frontend/pages/post/[id].tsx:25]: Sorry I'm a bit confused about how this makes sense 🤔

File: frontend/public/images/icons/youtube.svg [hunk 30]
```diff
@@ -0,0 +1,11 @@
...
+      <circle cx="30" cy="30" r="30" stroke="none"/>
+      <circle cx="30" cy="30" r="29" fill="none"/>
+    </g>
+    <g id="youtube-icon" transform="translate(429 537.912)">
+      <path id="Path_259" data-name="Path 259" d="M29.983,81.291q-.016-.9-.142-2.285a20.784,20.784,0,0,0-.36-2.469,3.893,3.893,0,0,0-1.163-2.059,3.568,3.568,0,0,0-2.084-.971A109.83,109.83,0,0,0,15,73.088a109.826,109.826,0,0,0-11.233.418,3.53,3.53,0,0,0-2.076.971A3.915,3.915,0,0,0,.536,76.537a18.529,18.529,0,0,0-.377,2.469Q.034,80.387.017,81.291T0,83.8q0,1.607.017,2.511T.159,88.6a20.707,20.707,0,0,0,.36,2.47,3.891,3.891,0,0,0,1.163,2.059,3.57,3.57,0,0,0,2.084.971A109.72,109.72,0,0,0,15,94.517,109.721,109.721,0,0,0,26.233,94.1a3.53,3.53,0,0,0,2.076-.971,3.916,3.916,0,0,0,1.155-2.059,18.541,18.541,0,0,0,.377-2.47q.125-1.381.142-2.285T30,83.8Q30,82.2,29.983,81.291Zm-9.057,3.415-8.571,5.357a.955.955,0,0,1-.569.167,1.158,1.158,0,0,1-.519-.134,1,1,0,0,1-.552-.937V78.445a1,1,0,0,1,.552-.937,1.008,1.008,0,0,1,1.088.033L20.926,82.9a1.065,1.065,0,0,1,0,1.808Z" fill="#fff"/>
+    </g>
+  </g>
+</svg>
```
Comment (robin-macpherson) [inline frontend/public/images/icons/youtube.svg:11]: Aren't we deleting the original icon SVGs after converting them to React components?

File: frontend/components/Dashboard/Posts/Post.tsx (hunk 27 above)
Comment (nlitwin) [reply frontend/components/Dashboard/Posts/Post.tsx:31]: This is just to get rid of the TypeScript errors for unused variables until the avatar part is implemented in the next PR 😁

File: frontend/pages/post/[id].tsx (hunk 29 above)
Comment (nlitwin) [reply frontend/pages/post/[id].tsx:25]: Doh! I think I confused myself ! I will revert this back to `Post` and then make the Posts that appear on the my feed / profile / my posts pages `PostCard`

**Comment by robin-macpherson** [comment]
@nlitwin just uploaded the SVGs for the 'likes' and 'comments' icons to the drive ✌🏼📨

File: backend/src/graphql.ts (hunk 25 above)
Comment (nlitwin) [reply backend/src/graphql.ts:161]: Since we already have a `feed` query that finds posts based on things other than `authorId`, I was trying to think of a use case when we'd want to use the `posts` query for purposes other than a feed and not have an `authorId`, but I couldn't come up with one. My posts, drafts, and the profile page all require an `authorId`. Perhaps it should be called `postByAuthorId` like we have `postById`? I'm open to suggestions! 😄

File: frontend/components/Dashboard/Profile/ProfileCard.tsx (hunk 22 above)
Comment (nlitwin) [reply frontend/components/Dashboard/Profile/ProfileCard.tsx:36]: Generally it's best to just use an eslint rule for this kind of thing so that it's just all the same, but I just tried to find an eslint rule for empty lines in HTML and I couldn't find one. Prettier also doesn't change empty lines on HTML files, so we shouldn't run into a situation where someone saves an HTML file and it auto-formats. I personally don't think we should have a rule around empty lines in HTML, no empty lines will make some files with dense information look really confusing, and enforcing empty lines for everything can make files bloated and lengthy. I think that's also why there isn't an eslint rule for this specifically, because it's an "it depends" situation on when to use empty lines for readability. In terms of a convention, my personal preference is just, "format it so it's readable"

File: frontend/components/Dashboard/PostCard/MyFeedPostCard.tsx (hunk 17 above)
Comment (nlitwin) [reply frontend/components/Dashboard/PostCard/MyFeedPostCard.tsx:79]: Yup yup! it's `PostCard` 😊

File: backend/src/graphql.ts [hunk 31]
```diff
@@ -317,42 +340,39 @@ schema.mutationType({
...
+
+        const isValid = await bcrypt.compare(args.password, user.auth.password)
+
+        if (!isValid) {
+          throw new Error('Invalid password')
+        }
+        const token = jwt.sign({ userId: user.id }, process.env.APP_SECRET!)
+        ctx.response.cookie('token', token, {
+          httpOnly: true,
+          maxAge: ONE_YEAR,
```
Comment (robin-macpherson) [inline backend/src/graphql.ts:371]: Here's at least one example of where merging this would be bring back in old stuff into master. `ONE_YEAR` has been removed entirely as mentioned in a comment up above. I am very worried about this merge, especially in this file and it's very hard to parse everything quickly with all the formatting changes and the way that they are... but this kind of thing (this example) is what I was worried about earlier when saying I am worried about this merge

File: backend/src/utils.ts [hunk 32]
```diff
@@ -1,5 +1,6 @@
 import escapeHTML from 'escape-html'
 import { User, Post, Comment } from '.prisma/client'
+import { readTime } from './utils/post'
```
Comment (robin-macpherson) [inline backend/src/utils.ts:3]: What I'm seeing in the diff of this file does not match what is currently on `master`
image

File: backend/src/utils/post.ts [hunk 33]
```diff
@@ -0,0 +1,6 @@
...
+  const numWords = text.split(' ').length
+
+  return Math.round(numWords / 200)
+}
```
Comment (robin-macpherson) [inline backend/src/utils/post.ts:6]: Is there any reason to not put this in the `utils` file that has all the other stuff around posts?

File: frontend/components/Dashboard/MyFeed/MyFeed.tsx [hunk 34]
```diff
@@ -1,13 +1,11 @@
 import React from 'react'
-
-import PostCard from '../PostCard'
+import { Post } from '../../../generated/graphql'
+import MyFeedPostCard from '../PostCard/MyFeedPostCard'
```
Comment (robin-macpherson) [inline frontend/components/Dashboard/MyFeed/MyFeed.tsx:3]: All of these references to `MyFeedPostCard` are 100% updated in the PR branched off of this one right?

File: frontend/pages/dashboard/profile.tsx [hunk 35]
```diff
@@ -0,0 +1,37 @@
...
+
+interface InitialProps {
+  namespacesRequired: string[]
+}
+
+const ProfilePage: NextPage<InitialProps> = () => {
+  const { loading, data, error } = usePostsQuery({
+    variables: {
+      // TODO(remove hardcoded value once currentUser is available on the application state)
+      authorId: 1,
```
Comment (robin-macpherson) [inline frontend/pages/dashboard/profile.tsx:16]: I thought this TODO would have been taken care of when you merged in `<AuthGate>` and updated accordingly no?

**Comment by nlitwin** [comment]
Closing in favor of smaller PRs
